kalaspuffar/secure-quick-reliable-login
# Add input validation to identity settings

Issue #457.

**Description:**

Add input validation to all `EditText` input fields in the identity settings activtiy. This fixes a bunch of issues which could be caused by either leaving input fields empty or entering very large values.

**Changes:**

- Before, the few validations present were done upon hitting the "Save" button. Now, input validation is done "on-the-fly", directly when entering the values.
- The "Save" button is now dynamically enabled/disabled according to the input validation state
- Input errors are displayed directly on the affected input field instead of in a popup message
- Remove the arbitrary _QuickPass Timeout_ limit of one week

The whole process should be much cleaner now.

**Validations that are performed now:**

- Check for empty field
- Check for non-numeric values
- Upper bounds checking

<img src="https://user-images.githubusercontent.com/4005543/73123016-5195d780-3f8b-11ea-9953-71732ba1ae95.png" width="300">
